R/table.R

```diff
@@ -0,0 +1,105 @@
+#' Generics for tabular vectors
+#'
+#' @description
+#'
+#' \Sexpr[results=rd, stage=render]{vctrs:::lifecycle("experimental")}
+#'
+#' These generics make it possible to work with tables, i.e. vectors
+#' of dimension 2 like data frames and matrices. The main motivation
+#' of this API is to support colwise operations such as slicing
+#' columns. Operating across columns requires slightly different
+#' semantics and concepts than rowwise operations:
+#'
+#' - A tabular prototype is a zero-rows, zero-cols table. In order for
```

Comment:
Hmmm, this implies that a `grouped_df` has to be able to record that it groups columns that it doesn't contain. That feels a bit weird.

Reply:
I think that's only the case for static grouped-df, which makes sense. A prototype of dynamic grouped-dfs doesn't refer to any grouping variable.

Reply:
But it has to list the columns that it would apply to? Otherwise where is that information stored?

Reply:
For static gdf the grouping information is stored in the _type_. For dynamic gdf the grouping information is stored in the _data_ (instrumented df-cols in the proxy).